### ISSUE TITLE
Errors when attempting to import an existing AKS cluster

### ISSUE THREAD

**Issue by polivbr** [rancher/terraform-provider-rancher2#771]
When attempting import an existing AKS cluster by creating a new AKS cluster with "imported" set to true, many unnecessary fields are populated and sent to Rancher, causing the Rancher server to error out as it does not allow changes to those properties during an import.

**Comment by rawmind0** [comment]
The example published at docs should work fine,  https://registry.terraform.io/providers/rancher/rancher2/latest/docs/resources/cluster#importing-aks-cluster-from-rancher-v2-using-aks_config_v2-for-rancher-v260-or-above-tech-preview, is  not working for you?? No other field should be set and the provider should not send any other fields to Rancher.

**Comment by polivbr** [comment]
That example doesn't work. Using this:

```
terraform {
  required_providers {
    rancher2 = {
      source  = "rancher/rancher2"
    }
  }
}

provider "rancher2" {
  api_url    = var.rancher2_api_url
  token_key  = var.rancher2_token_key
}

data "rancher2_cloud_credential" "cred" {
    name = "azure-infra"
}

resource "rancher2_cluster" "intra" {
  name = "infra"
  description = "Terraform AKS cluster"
  aks_config_v2 {
    name = "aks-infra"
    cloud_credential_id = data.rancher2_cloud_credential.cred.id
    resource_group = "rg-aks-infra"
    resource_location = "centralus"
    imported = true
  }
}

```

Rancher gives me the following error:

`failed to update cluster: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="PropertyChangeNotAllowed" Message="Changing property 'networkProfile.networkPlugin' is not allowed." Target="networkProfile.networkPlugin"
`

If I use a version of the provider using the associated PR I submitted, it imports without error.

**Comment by rawmind0** [comment]
That's weird, this example is working fine to me when importing an aks cluster at rancher v2.6.0 and v2.6.1. `network_plugin` shouldn't be set. Just curiosity, what provider and Rancher versions are you using??

Anyway, the code is failing at some point as also seen at https://github.com/rancher/terraform-provider-rancher2/issues/757 

Thanks for your PR. I've took a look to it and is fine it should address the issue, but it's incomplete. I've rebased  and completed your PR with https://github.com/rancher/terraform-provider-rancher2/pull/775 . Please, take a look.

**Comment by polivbr** [comment]
That is weird. I'm running Rancher 2.6.1 with provider version 1.20.1.

Thanks for fixing up my PR. I did enough to solve my problem but am not familiar enough with the codebase to make the other changes you made.

**Comment by rawmind0** [comment]
> That is weird. I'm running Rancher 2.6.1 with provider version 1.20.1.

Agreed, it's weird. May be it's depending on the aks cluster config to import?? Anyway, PR #775 is already merged and the fix will be available at next tf provider release, v1.21.0

> Thanks for fixing up my PR. I did enough to solve my problem but am not familiar enough with the codebase to make the other changes you made.

Sure, no problem. Thank you for reporting and your PR.
